native/src/main/java/io/ballerina/stdlib/oracledb/parameterprocessor/OracleDBResultParameterProcessor.java

```diff
@@ -247,47 +245,55 @@ private Object convertInterval(String interval, int sqlType, Type ballerinaType,
                 case TypeTags.RECORD_TYPE_TAG:
                     try {
                         if (sqlType == OracleTypes.INTERVALDS) {
-                            //format: DD HH:Min:SS.XXX
+                            //format: [-]DD HH:Min:SS.XXX
                             if (ballerinaType.getName().
-                                    equalsIgnoreCase(io.ballerina.stdlib.time.util.Constants.TIME_OF_DAY_RECORD)) {
-                                Time time = Time.valueOf(interval.split("\\s+")[1].split("\\.")[0]);
-                                return Utils.createTimeRecord(time);
-                            } else if (ballerinaType.getName().
                                     equalsIgnoreCase(Constants.Types.INTERVAL_DAY_TO_SECOND_RECORD)) {
+                                boolean isNegative = interval.startsWith("-");
```

Comment:
Shall we declare the `isNegative` variable once at the beginning of the function instead of re-declaring it or, simply use `interval.startsWith("-")` in all the `if` conditions.

Reply:
fixed in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/24/commits/e7d39803884a1fa2c9c5c257276113dc3bce2ec6